### ISSUE TITLE
Revise skip link route change behavior

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#15006]
## Short background / context

Due to the header caching in Forem, we need to manually handle route changes to ensure keyboard accessibility. Without handling this in any way, a keyboard user isn't really able to take advantage of our "Skip to content" link, because when the route changes, focus is always _after_ the header. This means users don't have a proper mechanism to bypass repeating blocks of content which is a WCAG requirement.

## What we currently do

The current pattern for route changes in our app is:

- New page is announced by screen reader via `aria-live` region
- "Skip to content" link is focused, resetting focus to the top of the page, ready to continue navigation
- We only visibly show the focused skip link if the user changed page using the keyboard (e.g. pressing `Enter` on a link)

This approach was adopted based on the extensive research conducted by Marcy Sutton with Fable Labs participants, documented in their post: [What we learned from user testing of accessible client-side routing techniques with Fable Tech Labs](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing/)

## Why it needs to change

When we implemented this pattern we didn't fully appreciate how strange it would make the behaviour on the search page. Users very often submit search forms by `Enter` press,  even if they otherwise use a mouse or touch device to get around the app. This means that for the vast majority of users, the skip link is popping up when they don't expect it.

This is disorientating, and in the case of mobiles and small screens it also has the effect of blocking/covering otherwise useful header content/links. This is because of the way we absolutely position the skip link above everything else.

See related bug reports:

https://github.com/forem/forem/issues/13876
https://github.com/forem/forem/pull/14999

## Balancing concerns

Unfortunately, there isn't a "one fix that makes everything great for all users". The research linked above dives into this in more detail, but as a short summary for our use case, our wish list is:

1. Focus is reset to the top of the page so that keyboard navigation, and associated skip link, is usable
2. Focus should be predictably on an interactive element
3. For people who rely on visible focus indicators, the current focus is always visible
4. Screen reader users are made aware of the route change
5. Users who otherwise wouldn't be aware of the skip link, don't see it when they submit a search form, regardless of whether they submitted the form by pressing `Enter` or not.

## Suggested change

It's not 100% ideal, but I think we can have 3 out of 5 of our wish list by:

- Keeping the current `aria-live` announcement
- Adding a focusable (via `tabindex="-1"`) element that is hidden to sighted users as the first element on any page
- Focusing that new element on route change. 

This would mean that:

- For keyboard users, next time they press `Tab`, the 'skip to content' link would be focused and visible
- For screen reader users, the route change would be announced, but a "random" item would receive focus (potentially confusing, but I think at least partly mitigated by the route change announcement)
- For sighted users who don't use any assistive tech, it looks and feels like nothing is happening 😄 
- For users who use speech recognition or screen magnifying software etc, it may quite be disorientating because they won't be able to see where the focus has gone, but at least the page will have reset scroll position back to the top which might give _some_ clue

## Additional notes

What we have learned from the previous approach is that when we do implement this, or any other pattern, we need to test it out on touch devices and small viewports as well as desktop.
